cmd/openshift/operator/kodata/tekton-addon/addons/02-clustertasks/source_local/buildah/buildah-task.yaml

```diff
@@ -71,10 +78,21 @@ spec:
     image: $(params.BUILDER_IMAGE)
     workingDir: $(workspaces.source.path)
     script: |
-      buildah --storage-driver=$(params.STORAGE_DRIVER) bud \
-        $(params.BUILD_EXTRA_ARGS) --format=$(params.FORMAT) \
-        --tls-verify=$(params.TLSVERIFY) --no-cache \
-        -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
+      if [[ "$(workspaces.rhel-entitlement.bound)" == "true" ]]; then 
+        mkdir /tmp/entitlement
+        cp $(workspaces.rhel-entitlement.path)/entitlement.pem /tmp/entitlement
+        cp $(workspaces.rhel-entitlement.path)/entitlement-key.pem /tmp/entitlement
+      
+        buildah --storage-driver=$(params.STORAGE_DRIVER) bud \
+          $(params.BUILD_EXTRA_ARGS) --volume /tmp/entitlement:/etc/pki/entitlement --format=$(params.FORMAT) \
+          --tls-verify=$(params.TLSVERIFY) --no-cache \
+          -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT) 
+      else
+        buildah --storage-driver=$(params.STORAGE_DRIVER) bud \
+          $(params.BUILD_EXTRA_ARGS) --format=$(params.FORMAT) \
+          --tls-verify=$(params.TLSVERIFY) --no-cache \
+          -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
+      fi
```

Comment:
@savitaashture please do not duplicate the command. if entitlement used, update a variable and use that.

Reply:
Yes agree here, we can remove duplication of command, and better use variable to provide extra argument

Reply:
Yup make sense
Updated